picker update

w01fgang opened the pull request.

0455482 committed:
picker update

0455482 commented:
1. In Calendar.js class in the selectDate function I added if / else statement to check if the returnState is false, and if its false i call the onSelect function that changes the returnState variable so i can select the dateTo without switching tab. After that when we select the second date, it goes in the else section where i call the function switchTab to switch the tab to the time picker, and the function onSelect to change the returnState to false (dateFrom) so i after that i can pick time for dateFrom.

2. In the DateTimePicker class in the function handleConfirmClick I removed the seting of the tab to 0, so i can add the onSelect function to change the returnState when picking time. After that i am checking if the returnState is true (dateTo) if thats true I'm calling the function onClose to close the modal. Also i binded onSelect function with the Calendar component.

0455482 committed:
update example

w01fgang commented:
Sorry for the delay.
I still see in the example single month selector, I expected to see similar to this:
![expected](https://user-images.githubusercontent.com/8686751/59151410-e1a20380-8a32-11e9-9575-b1a49a0ceb7a.png)

w01fgang commented on example/src/App.js:
```diff
@@ -1,113 +1,206 @@
+    this.setDate = this.setDate.bind(this)
+    this.togglePicker = this.togglePicker.bind(this)
+    this.changeDateStatus = this.changeDateStatus.bind(this)
+    this.changeTimeStatus = this.changeTimeStatus.bind(this)
```
you don't need to bind methods as they already bound by `handleSelect = () => {}`

w01fgang commented on example/src/App.js:
```diff
@@ -1,113 +1,206 @@
-    });
+  changeTimeStatus = () => {
+    console.log("change time state")
+    this.setState({timeState: 'timeTo', dateState: 'dateFrom'})
```
where do you use `timeState`?

w01fgang commented on example/src/App.js:
```diff
@@ -1,113 +1,206 @@
+          onClose={this.togglePicker}
           language="en"
-          label={returnState ? 'Return' : 'Get'}
+          label="Some Label"
```
The label explains which date the user is currently selecting.

0455482 committed:
double date update

1. updated example, removed all the commented code.
2. In Calendar.js added some style, changed the function for navigating between months, added logic for one month days in one calendar, made container for one calendar and a button, and component with the 2 calendars.
3.DateTimePicker.js:  added some style, TimeContainer for the two time pickers, added state for selectedTimer, and button on each timer to select the time.
4. Day.js: added null days for the void spaces in the calendar.
5. getDate.js: updated just for next month
6. Modal.js: added style
7. Time.js: added style, updated the functions for hours, minuts (up, down), so they can change just the value, added button to confirm the time

0455482 committed:
fixes

0455482 committed:
fixes

Fixed the bug that we talked about. Also fixed, when you go to next month not to change the selected date.

0455482 committed:
fix

0455482 committed:
fix

fixed bug when chossing 31 dec. 2018...

0455482 committed:
fix

fixed the jumping buttons when switching months, and added year check at the selected checking.

w01fgang committed:
🏷️ add flow types

w01fgang committed:
:arrow_up: update deps

w01fgang committed:
:recycle: refactor & fix lint and flow errors

w01fgang committed:
:construction: WIP

w01fgang committed:
:construction: WIP

w01fgang committed:
:construction: WIP

w01fgang committed:
:construction: WIP

w01fgang committed:
:wrench: update flow config

w01fgang committed:
:construction: WIP

w01fgang committed:
:arrow_up: update deps

w01fgang committed:
:wrench: ignore example for npm module

w01fgang committed:
:recycle: refactor day/calendar

w01fgang committed:
:fire: remove console logs

w01fgang committed:
:bug: remove duplicated dependency

w01fgang committed:
:recycle: refactor render

w01fgang committed:
:arrow_up: update dependency in example

w01fgang committed:
:bug: fix disabled days

w01fgang committed:
:lipstick: update buttons styles

w01fgang committed:
:bug: fix month name on button

w01fgang committed:
:bug: fix month switch
